saleor/warehouse/migrations/0016_stock_quantity_allocated.py

```diff
@@ -0,0 +1,34 @@
+# Generated by Django 3.2.6 on 2021-12-21 11:33
+
+from django.db import migrations, models
+from django.db.models import Sum
+from django.db.models.functions import Coalesce
+
+
+def assign_quantity_allocated_to_stocks(apps, schema_editor):
+    Stock = apps.get_model("warehouse", "Stock")
+
+    for stock in Stock.objects.iterator():
+        if quantity_allocated := stock.allocations.aggregate(
+            quantity_allocated=Coalesce(Sum("quantity_allocated"), 0)
+        )["quantity_allocated"]:
+            stock.quantity_allocated = quantity_allocated
+            stock.save(update_fields=["quantity_allocated"])
```

Comment:
I'm not sure about this save in the for loop. For big databases, it's gonna take forever. IMO we should prepare the list of stock objects to update and then use `bulk_update` with `batch_size` defined.

Reply:
Preparing a list to update would kill it even further since this field is freshly introduced and that list would contain every stock from dataset. I've changed it to full db operation.
https://explain.dalibo.com/plan/tAR
test DB is not a real-world example since that contains 20 stocks and 100k allocations (one of these stocks have so much), though it shows quite straight forward execution plan. On that test local db, improvement was from 2minutes to 2.46s.
CC @fowczarek